core/server/worker/src/test/java/alluxio/worker/block/MonoBlockStoreCommitBlockTest.java

```diff
@@ -0,0 +1,142 @@
+
+// This Test is a little different from the PagedBlockStoreCommitStore due to structure different.
+// MonoBlockStore.commitBlock() will call TieredBlockStore.commitBlocked() first as commitLocal, then will call BlockMasterClient.commitBlock() as commitMaster
+// TieredBlockStore.commitBlock() call TieredBLockStore.commitBlockInternal inside them wake the EventListener for listener.onCommitToLocal()
+// MonoBlockStore will wake the EventListener for listener.onCommitToMaster after BlockMasterClient.commitBlock() successes
+// In a nutshell two onCommit events weren't called in same domain
+public class MonoBlockStoreCommitBlockTest {
+    public MonoBlockStore mMonoBlockStore;
+    BlockMasterClientPool mockedBlockMasterClientPool;
+    BlockMasterClient mockedBlockMasterClient;
+    BlockMetadataManager mBlockMetadataManager;
+    BlockLockManager mBlockLockManager;
+    TieredBlockStore mTieredBlockStore;
+    private static final String FIRST_TIER_ALIAS = TieredBlockStoreTestUtils.TIER_ALIAS[0];
+    private StorageDir mTestDir1;
+    /** Rule to create a new temporary folder during each test. */
+    @Rule
+    public TemporaryFolder mTestFolder = new TemporaryFolder();
+
+    final Long sessionId = 1L;
+    final Long blockId = 2L;
+    int FIRST_TIER = 1;
+    // Maybe location should be asserted as well.
+    BlockStoreEventListener mListener;
+    @Before
+    public void setup() throws Exception {
+        File tempFolder = mTestFolder.newFolder();
+        TieredBlockStoreTestUtils.setupDefaultConf(tempFolder.getAbsolutePath());
+
+        mockedBlockMasterClientPool = mock(BlockMasterClientPool.class);
+        mockedBlockMasterClient = mock(BlockMasterClient.class);
+        when(mockedBlockMasterClientPool.acquire()).thenReturn(mockedBlockMasterClient);
+        doNothing().when(mockedBlockMasterClientPool).release(any());
+        mBlockLockManager = new BlockLockManager();
+        mBlockMetadataManager = BlockMetadataManager.createBlockMetadataManager();
+
+        mTestDir1 = mBlockMetadataManager.getTier(FIRST_TIER_ALIAS).getDir(0);
+
+        mListener = mock(BlockStoreEventListener.class);
+        doAnswer((i) -> {
+            assertEquals(2L, i.getArguments()[0]);
+            return 0;
+        }).when(mListener).onCommitBlockToLocal(anyLong(), any(BlockStoreLocation.class));
+        doAnswer((i) -> {
+            assertEquals(2L, i.getArguments()[0]);
+            return 0;
+        }).when(mListener).onCommitBlockToMaster(anyLong(), any(BlockStoreLocation.class));
+    }
+
+    @Test
+    public void CommitLocalandCommitMasterBothSuccess() {
+        mTieredBlockStore = new TieredBlockStore(mBlockMetadataManager, mBlockLockManager);
+
+        prepareBlockStore();
+
+        mMonoBlockStore.commitBlock(sessionId, blockId, false);
+
+        verify(mListener).onCommitBlockToLocal(anyLong(), any(BlockStoreLocation.class));
+        verify(mListener).onCommitBlockToMaster(anyLong(), any(BlockStoreLocation.class));
```

Comment:
Since you are verifying that the `onCommitBlockToLocal` method is called, you need to verify that the method was called with the actual arguments
```suggestion
        verify(mListener).onCommitBlockToLocal(BLOCK_ID, any(BlockStoreLocation.class));
        verify(mListener).onCommitBlockToMaster(BLOCK_ID, any(BlockStoreLocation.class));
```

Reply:
I think here is expecting a matcher but actual argument, the verification of the argument is defined in the corresponding `doAnswer()`.